### DISCUSSION TITLE
Clarification on `toStateData` merging into a nested object field

### DISCUSSION THREAD

**Discussion by giorgosera** [serverlessworkflow/specification#694] [Q&A] [answered]
Hi there, let's say I have the following state object

```
{
    "name": "someStateName",
    "type": "operation",
    "actions": [{
      "functionRef": {
        "refName": "someRefName",
        "arguments": {
          "text": "Hello there"
        },
        "actionDataFilter": {
          "toStateData": "${ .parent.child }"
        }
      }
    }],
    "end": true
  }
  ```

Is it correct to use `toStateData` with a nested object like the above? In this case my intention is to merge the results of the action into `child` which is itself a field of object `parent`.

Scanning the examples and docs I could only find one example with such a format (most examples show merging into one level) so I thought to ask if my assumption is correct.

PS: The example I am referring to might have an error. This is the example [here](https://github.com/serverlessworkflow/specification/tree/main/examples#filling-a-glass-of-water). The `toStateData` looks like this: `".counts.current"`. I guess the correct one would be `${ .counts.current }`?

**Comment by cdavernas** [comment, marked as answer]
> Is it correct to use toStateData with a nested object like the above?

@giorgosera if your expression language allows it, it's perfectly correct to use such filter. JQ allows that out of the box AFAIK, so you're good there.

> PS: The example I am referring to might have an error. This is the example [here](https://github.com/serverlessworkflow/specification/tree/main/examples#filling-a-glass-of-water). The toStateData looks like this: ".counts.current". I guess the correct one would be ${ .counts.current }?

It's not an error. The spec mandates using the $ and mustaches prefix/suffix only in non-native runtime expression values. It should imo enforce it everywhere, though. 

Anyways, back to the sample: the `toStateData` property is a runtime expression, therefore the format does not forcibly applies.

**Reply by giorgosera** [reply]
Got it! Thanks 🙏
